hal/stm32f373/adc.c

```diff
@@ -178,6 +178,7 @@ void adc_trace(adc_channel_t *ch, volatile int16_t *dst, int count, int trigger,
 	if (adc->dma == NULL)
 		///@todo error current implementation does not support interrupts so we need a dma
 		return;
+	dma_cancel(adc->dma); // cancel any pending/running dma
```

Comment:
I was tempted to make the `dma_request()` function implement this logic so all DMA requests for all things like PWM, TMR, UART etc work this way ... but I was worried this might break something (it would require more testing than I have time for here) ... also there might be a use case to no cancel a running DMA request so in that case adding this logic to the other modules using the DMA is more flexible